### PR TITLE
Change linspace creation to match numpy when num equal to 0

### LINKED ISSUE #8670: `linspace` division by zero error
**What happened**:

`linspace` gives a division by zero error, when NumPy returns an empty array on the same input.

**What you expected to happen**:

Behaviour the same as NumPy.

**Minimal Complete Verifiable Example**:

```python
>>> import numpy as np
>>> np.linspace(0, 0, 0, endpoint=False)
array([], dtype=float64)
>>> import dask.array as da
>>> da.linspace(0, 0, 0, endpoint=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/tom/projects-workspace/dask/dask/array/creation.py", line 303, in linspace
    step = float(range_) / div
ZeroDivisionError: float division by zero
```

**Anything else we need to know?**:

This fix is needed for the Array API (https://github.com/dask/community/issues/109).

**Environment**:

- Dask version: main
- Python version: 3.8
- Operating System: Mac
- Install method (conda, pip, source): source

### PR DESCRIPTION
**Pull request by peterpandelidis**
- [x] Closes #8670
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by tomwhite** [comment]
Thanks for the fix @peterpandelidis. There's another case that would be worth fixing too: if `num` is 1.

```python
>>> import numpy as np
>>> np.linspace(0, 0, 1)
array([0.])
>>> import dask.array as da
>>> da.linspace(0, 0, 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/tom/projects-workspace/dask/dask/array/creation.py", line 303, in linspace
    step = float(range_) / div
ZeroDivisionError: float division by zero
```

**Comment by peterpandelidis** [comment]
Made a change for the above. Not the most elegant solution, but it works.

**Comment by jakirkham** [comment]
@tomwhite do you have thoughts on the change above?

**Comment by jsignell** [comment]
ok to test

File: dask/array/creation.py [hunk 1]
```diff
@@ -299,7 +299,7 @@ def linspace(
 
     range_ = stop - start
 
-    div = (num - 1) if endpoint else num
+    div = (num - 1) if (endpoint and num != 1) or num == 0 else num
```
[the remark this answers is not in the document]
Comment (jsignell) [reply dask/array/creation.py:302]: I find that first statement hard to read, so yeah I like your suggestion.

[the remark this answers is not in the document]
Comment (pavithraes) [reply dask/array/creation.py:302]: @peterpandelidis What do you think about this suggestion?

[the remark this answers is not in the document]
Comment (peterpandelidis) [reply dask/array/creation.py:302]: Yeah, I agree with the suggestion.
I've been away for a bit so haven't updated the PR. 
Will update tonight.

**Review by tomwhite — APPROVED** [review]
LGTM. Thanks for updating this @peterpandelidis.

**Comment by jsignell** [comment]
Thanks for working on this @peterpandelidis and thanks @tomwhite for reviewing!

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**